code/game/objects/items/indoctrinator.dm

```diff
@@ -0,0 +1,67 @@
+/obj/item/indoctrinator
+	name = "indoctrinator"
+	desc = "What!? Who gave you that? It is NOT POSSIBLE!!!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "indoctrinator"
+	item_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	force = 0
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	item_flags = NOBLUDGEON
+	var/given_faction = "Coders" // What faction it gives and checks for.
+	var/item_charges = 999 // How many uses left
+	var/indoc_message = "You are now part of coders. You will code funny PRs." // The message that will show up and added to the notes on use.
+	var/use_time = 4 SECONDS // How long will be the do_after.
+
+/obj/item/indoctrinator/examine(mob/user)
+	. = ..()
+	. += "It has [item_charges] uses left."
+
+/obj/item/indoctrinator/attack(mob/living/carbon/human/M, mob/living/carbon/human/user)
+	if(M == user)
+		to_chat(user, "<span class='warning'>You can't use [src] on yourself!</span>")
+		return
+	if(item_charges <= 0) // No charges left.
+		to_chat(user, "<span class='warning'>[src] has no uses left!</span>")
+		return
+	if(given_faction in M.faction) // Target already has the faction.
+		to_chat(user, "<span class='notice'>[M] is already a part of [given_faction].</span>")
+		return
+	if(M && (M.stat == DEAD || M.stat == UNCONSCIOUS))
+		to_chat(user, "<span class='warning'>[M] should be healthy to receive treatment!</span>")
+		return
+	if(!M.mind)
+		to_chat(user, "<span class='warning'>[M] has no mind!</span>")
+		return
+	if(!M.key)
+		to_chat(user, "<span class='warning'>[M] isn't sentient!</span>")
+		return
+	if(!M.client)
+		to_chat(user, "<span class='warning'>[M] isn't capable of receiving treatment at the moment.</span>")
+		return
+	if(!M.getorgan(/obj/item/organ/brain))
+		to_chat(user, "<span class='warning'>[M] is lacking a brain to operate on!</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin to place [src] near the head of [M].</span>")
+	if(!do_after(user, use_time, target = M))
+		return
```

Comment:
From the moment the timer begins and onwards you want to stop the attack chain, as you've already started an actual action.
You can put a `. = STOP_ATTACK_PROC_CHAIN` at this point for that to happen (before the do after).

Reply:
Also make use of `do_afters` or `targeted_by` to avoid multiple actions.